solidity/core/contracts/validator-manager/OutboxValidatorManager.sol

```diff
@@ -49,25 +82,150 @@ contract OutboxValidatorManager is MultisigValidatorManager {
     // ============ External Functions ============
 
     /**
-     * @notice Determines if a quorum of validators have signed an improper checkpoint,
+     * @notice Determines if a quorum of validators have signed an invalid checkpoint,
+     * failing the Outbox if so.
+     * A checkpoint is invalid if it commits to anything other than contiguous non-empty leaves
+     * from leaf index zero to the leaf index of the checkpoint.
+     * @dev Invalid checkpoints signed by individual validators are not handled to prevent
+     * a single byzantine validator from failing the Outbox.
+     * @param _outbox The outbox.
+     * @param _signedRoot The root of the signed checkpoint.
+     * @param _signedIndex The index of the signed checkpoint.
+     * @param _signatures Signatures over the checkpoint to be checked for a validator
+     * quorum. Must be sorted in ascending order by signer address.
+     * @param _invalidLeaf The differing element in the fraudulent tree.
+     * @param _invalidProof Proof of inclusion of `_fraudulentLeaf`.
+     * @param _invalidIndex The index of the disputed leaf.
+     * @return True iff invalidity was proved.
+     */
+    function invalidCheckpoint(
+        IOutbox _outbox,
+        bytes32 _signedRoot,
+        uint256 _signedIndex,
+        bytes[] calldata _signatures,
+        bytes32 _invalidLeaf,
+        bytes32[32] calldata _invalidProof,
+        uint256 _invalidIndex
+    ) external returns (bool) {
```

Comment:
I don't feel strongly. Was thinking it would be nice to be able to make a static call to this function off-chain to verify that the call you're making will succeed, and being able to check that the call returns true is slightly nicer than checking that it doesn't revert.